src/query/plan/physical_test.go

```diff
@@ -70,34 +65,77 @@ func TestResultNode(t *testing.T) {
 }
 
 func TestShiftTime(t *testing.T) {
-	fetchTransform := parser.NewTransformFromOperation(functions.FetchOp{}, 1)
-	agg, err := aggregation.NewAggregationOp(aggregation.CountType, aggregation.NodeParams{})
-	require.NoError(t, err)
-	countTransform := parser.NewTransformFromOperation(agg, 2)
-	transforms := parser.Nodes{fetchTransform, countTransform}
-	edges := parser.Edges{
-		parser.Edge{
-			ParentID: fetchTransform.ID,
-			ChildID:  countTransform.ID,
+	tests := []struct {
+		name             string
+		fetchOp          functions.FetchOp
+		lookbackDuration time.Duration
+		step             time.Duration
+		wantShiftBy      time.Duration
+	}{
+		{
+			name:             "by lookbackDuration",
+			fetchOp:          functions.FetchOp{},
+			lookbackDuration: 15 * time.Minute,
+			step:             time.Second,
+			wantShiftBy:      15 * time.Minute,
+		},
+		{
+			name:             "shift by range + lookbackDuration",
+			fetchOp:          functions.FetchOp{Offset: time.Minute, Range: time.Hour},
+			lookbackDuration: 5 * time.Minute,
+			step:             time.Second,
+			wantShiftBy:      time.Hour + 5*time.Minute, // range + max(offset, lookbackDuration)
+		},
+		{
+			name:             "shift by range + offset",
+			fetchOp:          functions.FetchOp{Offset: 8 * time.Minute, Range: time.Hour},
+			lookbackDuration: 5 * time.Minute,
+			step:             time.Second,
+			wantShiftBy:      time.Hour + 8*time.Minute, // range + max(offset, lookbackDuration)
```

Comment:
I think this should be 13 right? Offset just moves the series back by 8 minutes, should do the same for the start of the lookback period

Reply:
Makes sense, will change to always add `LookbackDuration` to the shift, in addition to `maxOffset` and `maxRange`.

Reply:
I think it may actually have to be `max(lookback, range)`

Reply:
Hmm... actually, if range is set, [we should just use range](https://github.com/prometheus/prometheus/blob/master/promql/engine_test.go#L284)

Reply:
Done.